app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,69 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, only: %i[new create]
+
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    @post = Post.new
+  end
+
+  # GET /posts/1/edit
+  def edit; end
+
+  # POST /posts
+  # POST /posts.json
+  def create
+    @post = Post.new(title: params[:post][:title], body: params[:post][:body], user_id: current_user.id)
```

Comment:
You should use strong parameters instead of passing them as you are doing in the create method. Check this [article](https://api.rubyonrails.org/classes/ActionController/StrongParameters.html) to read more about it.
The method you create has to be private to secure the information.

You are not validating the post correctly. I can create an empty or incomplete post and that shouldn't happen.

**Optional**: the way you are creating a new post is not the best. You could make use of the `current_user` method provided by `Devise` and do something like this: `current_user.posts.build(post_params)` That way, the user is automatically added to the post in the DB.